12.0 add product supplier sale price

MPoch-PlanetaTIC opened the pull request:
Set Sale Price using prices suggested by supplier

MPoch-PlanetaTIC committed:
[ADD] product_supplier_sale_price: Suppliers can suggest a sale price

MPoch-PlanetaTIC committed:
[FIX]product_supplier_sale_price: Fix product.supplierinfo view layout.

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_price: Added readme files.

rousseldenis commented on product_supplier_sale_price/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+    'category': 'Product',
+    'summary': "Set Sale Price using prices suggested by supplier",
+    'author': 'PlanetaTIC',
+    'website': 'https://www.planetatic.com',
```
Please change according to https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10

rousseldenis commented on product_supplier_sale_price/__manifest__.py:
```diff
@@ -0,0 +1,23 @@
+    'version': '12.0.1.0.0',
+    'category': 'Product',
+    'summary': "Set Sale Price using prices suggested by supplier",
+    'author': 'PlanetaTIC',
```
Please complete this : https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11

rousseldenis commented on product_supplier_sale_price/models/product.py:
```diff
@@ -0,0 +1,148 @@
+        return res
+
+
+class ProductProduct(models.Model):
```
Please put one model per file https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#file-naming

rousseldenis commented on product_supplier_sale_price/models/product.py:
```diff
@@ -0,0 +1,148 @@
+                super(ProductProduct, product)._compute_lst_price()
+
+    @api.multi
+    def _prepare_supplier_sale_price(self, force=False):
```
I would have done this implementing multi recordset in order to improve performances.

rousseldenis commented on product_supplier_sale_price/models/product.py:
```diff
@@ -0,0 +1,148 @@
+    @api.multi
+    def _update_supplier_sale_price(self):
+        product_obj = self.env['product.product']
+        self.ensure_one()
```
See comment about multi implementation.

rousseldenis commented on product_supplier_sale_price/models/product.py:
```diff
@@ -0,0 +1,148 @@
+                sale_price = False
+            product.supplier_sale_price = sale_price
+
+    use_supplier_sale_price = fields.Boolean('Use Supplier Sale Price')
```
Please put fields on top of the file

rousseldenis commented on product_supplier_sale_price/models/product.py:
```diff
@@ -0,0 +1,148 @@
+                 'variant_seller_ids.sale_price')
+    def _compute_lst_price(self):
+        for product in self:
+            if product.use_supplier_sale_price:
```
I would have done two recordset in order to do the super() once

rousseldenis reviewed (changes requested):
@MPoch-PlanetaTIC Thanks for your contribution. Could you attend comments and fix travis ?

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_price: One file per model. Applying multi
recordset on compute field. And minor fixes for travis.

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_price: When updating fix_price from
product_product, also update product_template's list_price if it has
only one variant.

rousseldenis commented:
@MPoch-PlanetaTIC

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_sale: avoid to overwrite
use_supplier_sale_price if it has the same value.

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_price: Fix removed use_supplier_sale_price
update on write and create according list_price.

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_price: When deleting a product_supplierinfo,
if the affected product has no more suppliers, set
use_supplier_sale_price to False.

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_price: added list_price computation of
product_product

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_price: Only apply supplier_sale_price when
it is set.

MPoch-PlanetaTIC committed:
[FIX] product_supplier_price: Only apply supplier price when supplier
price is set also applies to list_price.

MPoch-PlanetaTIC committed:
[FIX] product_supplier_sale_price: When removing a product_supplierinfo
do not set use_supplier_sale_price to False.
